### PR TITLE
Refactor credits and supply migration

### PR DESCRIPTION
**Pull request by tomlinton**

### CODE REVIEW

**Review by DanielVF — APPROVED** [review]
LGTM!

**Comment by DanielVF** [comment]
@tomlinton, the comment line " //  0=NotSet, 1=OptOut, 2=OptIn" should be deleted, since that is no longer the implementation.

**Comment by tomlinton** [comment]
Removed! Enums are just uints starting from 0 so it was accurate, not really needed though.

**Comment by DanielVF** [comment]
Oops, somehow I read that as starting with a "-1" - guess my memory too over my eyes. :(

File: contracts/contracts/token/OUSD.sol [hunk 1]
```diff
@@ -387,53 +374,97 @@ contract OUSD is Initializable, InitializableToken, Governable {
...
+        // Increase non rebasing credits
         nonRebasingCredits = nonRebasingCredits.add(
             _creditBalances[msg.sender]
         );
-        nonRebasingSupply = nonRebasingSupply.add(balanceOf(msg.sender));
+
+        // Set fixed credits per token
         nonRebasingCreditsPerToken[msg.sender] = rebasingCreditsPerToken;
+
+        // Decrease rebasinbg credits, total supply remains unchanged so no
```
Comment (DanielVF) [inline contracts/contracts/token/OUSD.sol:463]: Typo, "rebasinbg"

**Review by franckc — APPROVED** [review]
Great work. LGTM

**Pull request merged**